src/sst/core/serialization/impl/serialize_utility.h

```diff
@@ -17,24 +17,205 @@
+/////////////////////////////////////////////////////////////////////////////////////////////////////
+
+namespace pvt_trivial {
+
+// Whether a type has no serialize_order() method
+template <class, class = void>
+struct has_no_serialize_order : std::true_type
+{};
+
+template <class T>
+struct has_no_serialize_order<T,
+    std::void_t<decltype(std::declval<T>().serialize_order(std::declval<class serializer&>()))>> : std::false_type
+{};
+
+// If it's not a trivially copyable, standard layout aggregate without a serialize_order() method, it needs to
+// be an integer, floating-point, enum or member object pointer, or one of the specializations listed later.
+// std::conjunction_v and std::disjunction_v are used to short-circuit and speed up instead of using && and ||.
+template <class C, bool = std::conjunction_v<std::is_aggregate<C>, std::is_trivially_copyable<C>,
+                       std::is_standard_layout<C>, has_no_serialize_order<C>>>
+constexpr bool is_trivially_serializable_v =
+    std::disjunction_v<std::is_arithmetic<C>, std::is_enum<C>, std::is_member_object_pointer<C>>;
+
+// glob_ts is convertible to any trivially serializable type
+struct glob_ts
+{
+    // The type of the conversion must be trivially serializable or the conversion fails
+    template <class C, class = std::enable_if_t<is_trivially_serializable_v<C>>>
+    operator C() const;
+};
+
+// Whether all fields of an aggregate type are trivially serializable
+template <class C, class, class = C>
+constexpr bool has_ts_fields = false;
+
+// Try to initialize N fields in an aggregate, with glob_ts() automatically converting to each field's
+// type. If conversion fails, it is not an error (SFINAE) and this specialization won't apply.
+template <class C, size_t... I>
+constexpr bool has_ts_fields<C, std::index_sequence<I...>, decltype(C { (I, glob_ts())... })> = true;
+
+// If it's a trivially copyable, standard layout aggregate, then all of its fields must be trivially serializable
+template <class C>
+constexpr bool is_trivially_serializable_v<C, true> = has_ts_fields<C, std::make_index_sequence<nfields<C>>>;
+
+// std::bitset is trivially serializable
+template <size_t N>
+constexpr bool is_trivially_serializable_v<std::bitset<N>, false> = true;
+
+// Complex numbers are trivially serializable
+template <class C>
+constexpr bool is_trivially_serializable_v<std::complex<C>, false> = true;
+
+#ifndef __STDC_NO_COMPLEX__
+template <>
+inline constexpr bool is_trivially_serializable_v<float _Complex, false> = true;
+
+template <>
+inline constexpr bool is_trivially_serializable_v<double _Complex, false> = true;
+
+template <>
+inline constexpr bool is_trivially_serializable_v<long double _Complex, false> = true;
+#endif
+
+// Other floating-point types not covered by std::is_floating_point
+template <>
+inline constexpr bool is_trivially_serializable_v<_Float16, false> = true;
+
+template <>
+inline constexpr bool is_trivially_serializable_v<__float128, false> = true;
```

Comment:
Looks like _Float16 isn't available on the compilers on our Linux testers and _float128 isn't available on our Mac testers.  We'll either need to remove these are put checks around them.